src/api/firebase.ts

```diff
@@ -253,9 +254,41 @@ export async function addItem(
 export async function updateItem(listPath: string, item: ListItem) {
 	const itemDocRef = doc(db, listPath, "items", item.id);
 
-	const updates: Pick<ListItem, "totalPurchases" | "dateLastPurchased"> = {
+	const lastUpdatedDate = item.dateLastPurchased
+		? item.dateLastPurchased
+		: item.dateCreated;
+
+	// Last estimated date of next purchase, or previous dateNextPurchased in whole number
+	const previousEstimate = getDaysBetweenDates(
+		lastUpdatedDate.toDate(),
+		item.dateNextPurchased.toDate(),
+	);
+
+	const daysSinceLastPurchased = getDaysBetweenDates(
+		new Date(),
+		lastUpdatedDate.toDate(),
+	);
+
+	const newDateNextPurchased = getFutureDate(
+		calculateEstimate(
+			previousEstimate,
+			daysSinceLastPurchased,
+			item.totalPurchases,
+		),
+	);
+
+	console.log("item date last purchased:", item.dateLastPurchased?.toDate());
+	console.log("previous estimate:", previousEstimate);
+	console.log("Original next purchase date:", item.dateNextPurchased.toDate());
+	console.log("New Next Purchase Date!", newDateNextPurchased);
+
+	const updates: Pick<
```

Comment:
Really cool to see you guys building on top of the the Pick Utility Type!!

Reply:
This was awesome to really see in action. After Tanner suggested it last week and understanding that it allows us pick what already established type information we have and utilize it here. While also making sure to continually keep the code type safe was fun to extend on.